pinot-plugins/pinot-stream-ingestion/pinot-kinesis/src/main/java/org/apache/pinot/plugin/stream/kinesis/server/KinesisDataServerStartable.java

```diff
@@ -0,0 +1,120 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.plugin.stream.kinesis.server;
+
+import cloud.localstack.Localstack;
+import cloud.localstack.ServiceName;
+import cloud.localstack.docker.annotation.LocalstackDockerConfiguration;
+import java.net.URI;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import org.apache.pinot.spi.stream.StreamDataServerStartable;
+import org.apache.pinot.spi.utils.NetUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.http.SdkHttpConfigurationOption;
+import software.amazon.awssdk.http.apache.ApacheSdkHttpService;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.kinesis.KinesisClient;
+import software.amazon.awssdk.services.kinesis.model.CreateStreamRequest;
+import software.amazon.awssdk.services.kinesis.model.DescribeStreamRequest;
+import software.amazon.awssdk.utils.AttributeMap;
+
+
+public class KinesisDataServerStartable implements StreamDataServerStartable {
+  private static final Logger LOGGER = LoggerFactory.getLogger(KinesisDataServerStartable.class);
+
+  public static final String NUM_SHARDS_PROPERTY = "numShards";
+  public static final String DEFAULT_REGION = "us-east-1";
+  public static final String DEFAULT_ACCESS_KEY = "access";
+  public static final String DEFAULT_SECRET_KEY = "secret";
+  public static final String DEFAULT_PORT = "4566";
+
+  private final Localstack _localstackDocker = Localstack.INSTANCE;
+  LocalstackDockerConfiguration _dockerConfig;
+  Properties _serverProperties;
+  private String _localStackKinesisEndpoint = "http://localhost:%s";
+
+  @Override
+  public void init(Properties props) {
+    _serverProperties = props;
+    final Map<String, String> environmentVariables = new HashMap<>();
+    environmentVariables.put("SERVICES", ServiceName.KINESIS);
+    _dockerConfig =
+        LocalstackDockerConfiguration.builder().portEdge(_serverProperties.getProperty("port", DEFAULT_PORT))
+            .portElasticSearch(String.valueOf(NetUtils.findOpenPort(4571))).imageTag("0.12.15")
+            .environmentVariables(environmentVariables).build();
+
+    _localStackKinesisEndpoint =
+        String.format(_localStackKinesisEndpoint, _serverProperties.getProperty("port", DEFAULT_PORT));
+  }
+
+  @Override
+  public void start() {
+    _localstackDocker.startup(_dockerConfig);
+  }
+
+  @Override
+  public void stop() {
+    _localstackDocker.stop();
+  }
+
+  @Override
+  public void createTopic(String topic, Properties topicProps) {
+    try {
+      KinesisClient kinesisClient = KinesisClient.builder().httpClient(
```

Comment:
can this code to create the clients (from here, from KinesisDataProducer and from KinesisConnectionHandler) be moved into a common ClientFactory under a utils package in this module? It'll help us be consistent, and also easily propagate changes

Reply:
I can do that but I will prefer it to do in a seperate PR. Reason being, it will affect KinesisConnectionHandler and KinesisIntegrationTest as well.

Reply:
makes sense 👍